// react-scroll-parallax/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "useController", {
  enumerable: true,
  get: function get() {
    return _useController2.default;
  }
});
Object.defineProperty(exports, "withController", {
  enumerable: true,
  get: function get() {
    return _withController2.default;
  }
});
Object.defineProperty(exports, "Parallax", {
  enumerable: true,
  get: function get() {
    return _Parallax2.default;
  }
});
Object.defineProperty(exports, "ParallaxProvider", {
  enumerable: true,
  get: function get() {
    return _ParallaxProvider2.default;
  }
});
Object.defineProperty(exports, "ParallaxBanner", {
  enumerable: true,
  get: function get() {
    return _ParallaxBanner2.default;
  }
});
Object.defineProperty(exports, "ParallaxContext", {
  enumerable: true,
  get: function get() {
    return _ParallaxContext2.default;
  }
});

var _useController2 = _interopRequireDefault(require("./components/useController"));

var _withController2 = _interopRequireDefault(require("./components/withController"));

var _Parallax2 = _interopRequireDefault(require("./components/Parallax"));

var _ParallaxProvider2 = _interopRequireDefault(require("./components/ParallaxProvider"));

var _ParallaxBanner2 = _interopRequireDefault(require("./components/ParallaxBanner"));

var _ParallaxContext2 = _interopRequireDefault(require("./helpers/ParallaxContext"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
