// @salesforce/design-system-react/module/components/progress-bar/__examples__/examples.js
import React from 'react';
import ProgressBar from '../../../components/progress-bar';
export var PROGRESS_0 = function PROGRESS_0() {
  return React.createElement(ProgressBar, {
    id: "progress-bar-1",
    value: 0
  });
};
export var PROGRESS_25 = function PROGRESS_25() {
  return React.createElement(ProgressBar, {
    id: "progress-bar-2",
    value: 25
  });
};
export var PROGRESS_HALF = function PROGRESS_HALF() {
  return React.createElement(ProgressBar, {
    id: "progress-bar-3",
    value: 50
  });
};
export var PROGRESS_75 = function PROGRESS_75() {
  return React.createElement(ProgressBar, {
    id: "progress-bar-4",
    value: 75
  });
};
export var COMPLETE_100 = function COMPLETE_100() {
  return React.createElement(ProgressBar, {
    id: "progress-bar-5",
    value: 100,
    color: "success"
  });
};
export var COMPLETE_WITH_DESCRIPTION = function COMPLETE_WITH_DESCRIPTION() {
  return React.createElement(ProgressBar, {
    value: 100,
    color: "success",
    label: "Einstein Setup Assistant"
  });
};
export var ROUNDED_BAR = function ROUNDED_BAR() {
  return React.createElement(ProgressBar, {
    id: "progress-bar-circular",
    value: 20,
    radius: "circular"
  });
};
export var X_SMALL_BAR = function X_SMALL_BAR() {
  return React.createElement(ProgressBar, {
    id: "progress-bar-x-small",
    value: 20,
    thickness: "x-small"
  });
};
export var SMALL_BAR = function SMALL_BAR() {
  return React.createElement(ProgressBar, {
    id: "progress-bar-small",
    value: 20,
    thickness: "small"
  });
};
export var MEDIUM_BAR = function MEDIUM_BAR() {
  return React.createElement(ProgressBar, {
    id: "progress-bar-medium",
    value: 20,
    thickness: "medium"
  });
};
export var LARGE_BAR = function LARGE_BAR() {
  return React.createElement(ProgressBar, {
    id: "progress-bar-large",
    value: 20,
    thickness: "large"
  });
};
export var VERTICAL_BAR = function VERTICAL_BAR() {
  return React.createElement(ProgressBar, {
    id: "progress-bar-vertical",
    orientation: "vertical",
    value: 20,
    thickness: "large"
  });
};
